django_cloud_deploy/cli/prompt_manager.py

```diff
@@ -0,0 +1,689 @@
+            return
+
+        pass_prompt = prompt_new.PasswordPrompt(self.console)
+        msg = 'Enter a password for the Django superuser "{}"'.format(
+            self.args['django_superuser_login'])
+        self.console.tell('{} {}'.format(self.step, msg))
+        self.args[self.PARAMETER] = pass_prompt.prompt()
+
+    def validate(self, s: str):
+        pass_prompt = prompt_new.PasswordPrompt(self.console)
+        return pass_prompt.validate(s)
+
+
+class DjangoSuperuserEmailPrompt(StringTemplatePromptManager):
+    """Allow the user to enter a Django email address."""
+
+    PARAMETER = 'django_superuser_email'
+    PARAMETER_PRETTY = 'Django superuser email'
+    DEFAULT_VALUE = 'test@example.com'
+
+    def validate(self, s: str):
+        """Validates that a string is a valid Django superuser email address.
+
+        Args:
+            s: The string to validate.
+
+        Raises:
+            ValueError: if the input string is not valid.
+        """
+        if not re.match(r'[^@]+@[^@]+\.[^@]+', s):
+            raise ValueError(('Invalid Django superuser email address "{}": '
+                              'the format should be like '
+                              '"test@example.com"').format(s))
+
+
+class RootPromptManager(object):
+    """Class at the top level that instantiates all of the Prompt Managers."""
+
+    PROMPT_ORDER = [
+        'credentials',
+        'project_id',
+        'project_name',
+        'billing_account_name',
+        'database_password',
+        'django_directory_path',
+        'django_project_name',
+        'django_app_name',
+        'django_superuser_login',
+        'django_superuser_password',
+        'django_superuser_email',
+    ]
+
+    required_parameters_to_prompt = {
+        'credentials': CredentialsPromptManager,
+        'project_id': GoogleProjectIdManager,
+        'project_name': GoogleProjectNameManager,
+        'billing_account_name': BillingPrompt,
+        'database_password': PostgresPasswordPrompt,
+        'django_directory_path': DjangoFilesystemPath,
+        'django_project_name': DjangoProjectNamePrompt,
+        'django_app_name': DjangoAppNamePrompt,
+        'django_superuser_login': DjangoSuperuserLoginPrompt,
+        'django_superuser_password': DjangoSuperuserPasswordPrompt,
+        'django_superuser_email': DjangoSuperuserEmailPrompt
+    }
+
+    def __init__(self, console: io.IO, args):
+        self.console = console
+        self.args = args
+
+    def _setup(self):
+        if self.args.get('use_existing_project', False):
+            self.args['project_creation_mode'] = (
+                workflow.ProjectCreationMode.MUST_EXIST)
+
+    def prompt(self):
+        self._setup()
+
+        total_steps = len(self.PROMPT_ORDER)
+        for i, prompt in enumerate(self.PROMPT_ORDER, 1):
+            step = '<b>[{}/{}]</b>'.format(i, total_steps)
+            self.required_parameters_to_prompt[prompt](self.console, step,
```

Comment:
So args is mutated through a shared dictionary. Yuck.

Reply:
probably will call it retrieve_parameters(..., args) -> args. Does that sound good?